cupy/cuda/compiler.py

```diff
@@ -538,7 +538,7 @@ def _convert_to_hip_source(source):
 
 
 def _compile_with_cache_hipcc(source, options, arch, cache_dir, extra_source,
-                              use_converter=True):
+                              backend='nvrtc', use_converter=True):
```

Comment:
Wouldn't it be nice to have a separate backend named "hiprtc"? The backend name shows up in the `CompileException` I think.
```suggestion
                              backend='hiprtc', use_converter=True):
```

Reply:
The problem I see here that I am not sure how to solve it in a clean way is that what would be the default value now.

Something like `"rtc"`, should we convert the nvrtc string to hiprtc before calling `compile_with_hipcc`?

Reply:
Maybe I missed something obvious, but shouldn't [my other comment](https://github.com/cupy/cupy/pull/3238#discussion_r400583951) addresses this?

Reply:
I am not sure I am following you, The backend default name is 'nvrtc', and so is the default value for the `backend` argument in the top-level function.
So if we change that for hip, we need to either do an 
```python
if backend == 'nvrtc'
   backend = 'hiprtc'
```
Or use an agnostic backend name for both such as `rtc`.

Reply:
Ah, let me guess what could be confusing. How about overwriting the backend in `compile_with_cache`? So, it'd be something like this:
```python
def compile_with_cache(..., backend='nvrtc', ...):
    ...
    if runtime.is_hip:
        backend = 'hiprtc' if backend == 'nvrtc' else 'hipcc'
        # name changed from "_compile_with_cache_hipcc" to "_compile_with_cache_hip"
        return _compile_with_cache_hip(..., backend=backend, ...)
    else:
        return _compile_with_cache_cuda(..., backend=backend, ...)
```

Reply:
I am not a big fan of doing this kind of things, but I'll add it since changing the global backend name leads to no-compat.

Reply:
Yeah I see your point. But I guess ad-hoc hacks like this would be inevitable unless we split the HIP support from `cupy.cuda`...